File: Java/find-minimum-in-rotated-sorted-array.java
//  Find Minimum in Rotate

// Time O(logN), Space O(1)
class Solution {
    public int findMin(int[] nums) {
        int l = 0, r = nums.length-1;
        while (l <= r) {
            if (nums[l] <= nums[r])
                break;
            int mid = l + (r-l) / 2;
            if (nums[mid] < nums[r])
                r = mid;
            else
                l = mid + 1;
        }
        return nums[l];
    }
}

File: leetcode/find-minimum-in-rotated-sorted-array.cc
//  Find Minimum in Rotated Sorted Array

class Solution {
public:
    int findMin(vector<int>& nums) {
        size_t l = 0, r = nums.size()-1;
        size_t mid = 0;
        while (l != r) {
            mid = l + (r - l) / 2;
            if (nums[mid] > nums[r])
                l = mid + 1;
            else
                r = mid;
        }
        return nums[l];
    }
};